File: 0001-two-sum/0001-two-sum.cpp
class Solution
{
    public:
        vector<int> twoSum(vector<int> &nums, int target)
        {
            map<int, int> mp;
            for (int i = 0; i < nums.size(); i++)
            {
                mp[nums[i]] = i;
            }
            for (int i = 0; i < nums.size(); i++)
            {
                int x = target - nums[i];
                if (mp[x] and mp[x] != i)
                {
                    return {
                        mp[x],
                        i
                    };
                }
            }
            return {};
        }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d={}
        for i in range(len(nums)):
            tempd={nums[i]:i}
            d.update(tempd)
        # print(d)
        for i in range(len(nums)):
            search=target-nums[i]
            # print(search)
            if search in d and i!=d[search]:
                return [i,d[search]]
        return []
                